src/main/java/core/mate/academy/producers/MachineProducer.java

```diff
@@ -0,0 +1,8 @@
+package core.mate.academy.producers;
+
+import java.util.List;
+
+public interface MachineProducer<T> {
+
```

Comment:
T should extend Machine, i think. Because you don't want other classes, like String or whatever to use this interface.